tests/test_butlerFits.py

```diff
@@ -0,0 +1,214 @@
+        self.assertFloatsAlmostEqual(
+            inputRecord.getShapeErr()[2, 2],
+            outputRecord.getShapeErr()[2, 2], rtol=1e-6)
+
+    def testFitsCatalog(self):
+        storageClass = self.storageClassFactory.getStorageClass("SourceCatalog")
+        datasetTypeName = "testCatalog"
+
+        catalog = self.makeExampleCatalog()
+        dimensions = self.butler.registry.dimensions.extract(["instrument", "visit"])
+        addDatasetType(self.butler, datasetTypeName, dimensions, storageClass)
+
+        dataId = {"visit": 42, "instrument": "DummyCam", "physical_filter": "d-r"}
+        ref = self.butler.put(catalog, datasetTypeName, dataId)
+        stored = self.butler.get(ref)
+        self.assertCatalogEqual(catalog, stored)
+
+    def testExposureCompositePutGetConcrete(self):
+        storageClass = self.storageClassFactory.getStorageClass("ExposureF")
+        ref = self.runExposureCompositePutGetTest(storageClass, "calexp")
+
+        uri = self.butler.getURI(ref)
+        self.assertTrue(os.path.exists(uri.path), f"Checking URI {uri} existence")
+
+    def testExposureCompositePutGetVirtual(self):
+        storageClass = self.storageClassFactory.getStorageClass("ExposureCompositeF")
+        ref = self.runExposureCompositePutGetTest(storageClass, "unknown")
+
+        primary, components = self.butler.getURIs(ref)
+        self.assertIsNone(primary)
+        self.assertEqual(set(components), COMPONENTS)
+
+    def runExposureCompositePutGetTest(self, storageClass, datasetTypeName):
+        example = os.path.join(TESTDIR, "data", "small.fits")
+        exposure = lsst.afw.image.ExposureF(example)
+        dimensions = self.butler.registry.dimensions.extract(["instrument", "visit"])
+        addDatasetType(self.butler, datasetTypeName, dimensions, storageClass)
+
+        assembler = ExposureAssembler(storageClass)
+
+        dataId = {"visit": 42, "instrument": "DummyCam", "physical_filter": "d-r"}
+        ref = self.butler.put(exposure, datasetTypeName, dataId)
+        # Get the full thing
+        composite = self.butler.get(datasetTypeName, dataId)
+
+        # There is no assert for Exposure so just look at maskedImage
+        self.assertMaskedImagesEqual(composite.maskedImage, exposure.maskedImage)
+
+        # Get each component from butler independently
+        for compName in COMPONENTS:
+            compTypeName = DatasetType.nameWithComponent(datasetTypeName, compName)
+            component = self.butler.get(compTypeName, dataId)
+
+            reference = assembler.getComponent(exposure, compName)
+            self.assertIsInstance(component, type(reference), f"Checking type of component {compName}")
+
+            if compName in ("image", "variance"):
+                self.assertImagesEqual(component, reference)
+            elif compName == "mask":
+                self.assertMasksEqual(component, reference)
+            elif compName == "wcs":
+                bbox = Box2I(Point2I(0, 0), Extent2I(9, 9))
```

Comment:
Does this need to be hardcoded, or could you use `exposure.getBBox`?

Reply:
Will take a look.